wled00/fcn_declare.h

```diff
@@ -186,10 +187,12 @@ class Usermod {
     virtual void readFromJsonState(JsonObject& obj) {}
     virtual void addToConfig(JsonObject& obj) {}
     virtual void readFromConfig(JsonObject& obj) {}
+    virtual void onMqttConnect(bool sessionPresent) {}
+    virtual bool onMqttMessage(char* topic, char* payload) { return false; }
     virtual uint16_t getId() {return USERMOD_ID_UNSPECIFIED;}
 };
 
-class UsermodManager {
+class UsermodManager : public Usermod {
```

Comment:
What is the reason you want `UsermodManager` to be a child class of `Usermod`?

Reply:
I was trying to get virtual functions propagate to UsermodManager (since I'm still not very good at OOP).
It has no added value and can be removed (to a previous no parent-child relationship).